src/fastga_he/powertrain_builder/powertrain.py

```diff
@@ -886,136 +886,128 @@ def _construct_connection_graph(self):
-                    propulsive_load_names.append(component_name)
-                # If there are gearboxes among neighbor, we also check for the neighbor of the
-                # gearbox. Not very generic way to do things :/
-                else:
-                    name_to_id = dict(zip(self._components_name, self._components_id))
-                    for neighbor in set(neighbors):
-                        if (
-                            name_to_id[neighbor] == "fastga_he.pt_component.speed_reducer"
-                            or name_to_id[neighbor] == "fastga_he.pt_component.planetary_gear"
-                            or name_to_id[neighbor] == "fastga_he.pt_component.gearbox"
-                        ):
-                            neighbors_gb = list(graph.neighbors(neighbor))
-                            if set(neighbors_gb).intersection(propulsor_name):
-                                propulsive_load_names.append(component_name)
-
-        distance_from_propulsive_load = {}
-        connections_length_between_nodes = dict(nx.all_pairs_shortest_path_length(graph))
+        :param references: Component type class or component type of reference component(s).
+                          Can be a single string or a list of type string entries.
+                          Each string should match a valid component type or component type class.
 
-        for component_name in self._components_name:
-            # When there are two separate sub propulsion chain in the same propulsion file,
-            # these line will cause an issue because, as it will browse all propulsive load he
-            # will attempt to reach loads he is not connected to and therefore not in
-            # connections_length_between_nodes. So first we must make sure to only browse
-            # connected loads.
+        :return: A dictionary mapping each component name to its minimum distance to any
+                 of the reference components. Distance is measured as the number of edges in
+                 the shortest path.
+        """
+        reference_component_types = []
 
-            connected_components = list(connections_length_between_nodes[component_name].keys())
-            connected_propulsive_loads = list(
-                set(propulsive_load_names) & set(connected_components)
-            )
+        # Packing single component type / component type class string into list
+        if isinstance(references, str):
+            references = [references]
 
-            min_distance = np.inf
-            for prop_load in connected_propulsive_loads:
-                distance_to_load = connections_length_between_nodes[component_name][prop_load]
-                if distance_to_load < min_distance:
-                    min_distance = distance_to_load
+        # Collect the reference component type by comparing the registered component type classes
+        # and component types to the given reference list
+        for id in resources.KNOWN_ID:
```

Comment:
id is not a valid variable name, it shadows built-in name from Python (Which PyCharm tells you btw ;) )